### PR TITLE
M3 1887 Release branch creation script and auto change log POC 

### PR DESCRIPTION
**Pull request by nwisnewski**
**Pupose:**
1. Wrote a generate change log python script, change log is based on git log diff between the release branch and master, the commit subject (title of PR request) is added into different sections by key words: breaking, added, changed, fixed. The commits titles added to the section are striped of the jira ticket numbers, the ticket numbers are then generated into a jql query and printed to the screen.

This would require the team to add keywords and ticket #'s in PR titles, we can work on what key words go in what section.

2. Added a script to cut the release branch, accepts new version and release date as the arguments, then cuts the branch updates yarn version, uses the script to auto generate change log and commits the changes. Then deletes the previous release branch from the repository and pushes the new release version.

### CODE REVIEW

File: scripts/release.sh [hunk 1]
```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+RELEASE=$1
```
Comment (patthiel) [inline scripts/release.sh]: Calling this `RELEASE_TAG` would be more descriptive

File: scripts/release.sh [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+#Get linode/manager repo name from remote branches
+GET_ORIGIN=$(git remote -v | grep git@github.com:linode/manager.git)
+set -- $GET_ORIGIN
+ORIGIN=$1
+
+#Create release branch
+git fetch $ORIGIN
+git checkout develop
+git git rebase $ORIGIN/develop
```
Comment (patthiel) [inline scripts/release.sh]: extra `git` here

File: scripts/release.sh [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+git checkout -b release-$RELEASE
+
+#Generate changelog
+python generate_changelog.py $RELEASE $RELEASE_DATE
+
+#Update Yarn versions
```
Comment (patthiel) [inline scripts/release.sh]: `Create release tag & bump package.json version` is a bit more accurate. This sounds like it's upgrading the version of `yarn` itself

File: generate_changelog.py [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+        if( 'test' in commit.lower()):
+            commit_array.pop(i)
+            break
+
+        if( 'breaking' in commit.lower()):
+            breaking.append(commit_array[i])
+            break
+
+        if( 'changed' in commit.lower()):
```
Comment (martinmckenna) [inline generate_changelog.py]: this is my own personal opinion and I'm not sure what the consensus is here, but I feel like these keywords should be present tense. And then be translated to past tense in the changelog.

**Pull request closed without merging**